src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1213,10 +1213,17 @@ def _need_update_container(self):
 
     def set_up_existing_container_update(self):
         if self.get_argument_force_single_container_updates():
+            containers = safe_get(self.new_containerapp, "properties", "template", "containers", default=[])
+            if not containers or len(containers) == 0:
+                raise ValidationError(
+                    "Cloud build image update failed. The container app '{}' does not have any containers.".format(self.get_argument_name()))
             # Remove n-1 containers where n is the number of containers in the containerapp and replace remaining container with the new container
             # Fails if all containers are removed
             while (len(self.new_containerapp["properties"]["template"]["containers"]) > 1):
                 self.new_containerapp["properties"]["template"]["containers"].pop()
+            # Set the container name to the container app name if force_single_container_updates is set
+            self.set_argument_container_name(self.get_argument_name())
```

Comment:
`self.set_argument_container_name(self.get_argument_name())` is redundant.

Reply:
@Greedygre are you referring to [this line](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/containerapp_decorator.py#L218)? If yes, this sets the container name to the name of the first container, so we need to re-set this in cloud build scenarios to make sure the container name is the container app name.

Reply:
Hi @snehapar9 

I am referring this Line:
https://github.com/Azure/azure-cli-extensions/blob/ccf5e9205a5a41f6a7c5d7f9fe69e9be3410693e/src/containerapp/azext_containerapp/custom.py#L1240

`container_name=name`

Reply:
@Greedygre thanks. The changes in this PR pops n-1 containers and replaces the last container with the source to cloud container. Since the container name is not set [here](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/containerapp_decorator.py#L508), we need to explicitly set it.